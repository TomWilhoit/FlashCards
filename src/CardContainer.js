import React, { Component } from "react";
import Card from "./Card";
import "./css/CardContainer.scss";

class CardContainer extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <div className="card-container">
        <div className="card">
          {this.props.questions.length !== 0 && (
            <Card
              id={this.props.questions[this.props.questionIndex].id}
              question={this.props.questions[this.props.questionIndex].question}
              correctAnswer={
                this.props.questions[this.props.questionIndex].correctAnswer
              }
              falseAnswer1={
                this.props.questions[this.props.questionIndex].falseAnswer1
              }
              falseAnswer2={
                this.props.questions[this.props.questionIndex].falseAnswer2
              }
              incrementQuestionIndex={this.props.incrementQuestionIndex}
              saveToStorage={this.props.saveToStorage}
              shouldRestart={this.props.shouldRestart}
              finishGame={this.props.finishGame}
            />
          )}
        </div>
      </div>
    );
  }
}

export default CardContainer;
