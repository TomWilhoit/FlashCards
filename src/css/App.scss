.App {
  text-align: center;
}

.review-header {
  display: flex;
  justify-content: center;
}

.question {
  display: flex;
  justify-content: center;
}
