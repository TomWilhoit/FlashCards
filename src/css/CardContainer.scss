

.card {
  display: grid;
  background-color: bisque;
  border-style: solid;
  margin: 25px;
  width: 80%;
  height: 30%;
};

.card-container {
  display: flex;
  justify-content: center;
  align-items: center;
  border-style: dashed;
}

.question {
  font-size: 40px;
}