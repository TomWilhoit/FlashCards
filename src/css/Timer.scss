.timer-clock {
  font-size: 70px;
  margin: 35px;
  color: crimson;
}