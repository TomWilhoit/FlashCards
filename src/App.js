import React, { Component } from 'react';
import CardCountainer from './CardContainer';
import Timer from './Timer';
import Banner from './Banner';
import Counter from './Counter';
import Card from './Card';
import FlashCardDataSet from './FlashCardDataSet';
import './css/App.css';

class App extends Component {
  constructor() {
    super();


    this.state = {
      questions: FlashCardDataSet,
      shouldRestart: false,
      questionIndex: 0,
      savedArray: [],
    }
  }
 
  shouldRestart = () => {
    if (this.state.shouldRestart===false) {
    this.setState({
      shouldRestart: true
    })
  } else{
    this.setState({
      shouldRestart: false
      })
    }
  }

  incrementQuestionIndex = () => {
    const newIndex = this.state.questionIndex + 1;
    this.setState({
      questionIndex: newIndex
    })
  }

  saveToStorage = () => {
    let [...savedQuestions] = this.state.savedArray
    savedQuestions.push(this.state.questions[this.state.questionIndex])
    localStorage.setItem('savedQuestions',JSON.stringify(savedQuestions))
    this.setState({
      savedArray: savedQuestions
    })

      

    }
  
  
  
  render() {
    return (
      <div className="App">
        <Banner/>
        <CardCountainer
        questionObj= {this.state.questions[this.state.questionIndex]}
        incrementQuestionIndex = {this.incrementQuestionIndex}
        saveToStorage = {this.saveToStorage}/>
        <Timer shouldRestart={this.shouldRestart}/>
        <Counter
        questionIndex = {this.state.questionIndex}/>

      </div>
    ) };
  }

export default App;
