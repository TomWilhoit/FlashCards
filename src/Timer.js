import React, { Component } from 'react';
import './css/App.css';
import App from './App';


export default class Timer extends Component {
  constructor(props) {
    super(props)
    
    this.state = {
      seconds: 900
    };
  }
  
  componentWillMount() {
    this.startTimer()
  }

  restartTimer = () => {
    this.setState({
      seconds: 900
    })
    this.props.shouldRestart();
  }
  
  startTimer = () => {
    this.timer = setInterval(this.eachTick, 1000);
  }
  

  eachTick = () => {
    if (this.state.seconds > 0)
    this.setState({
      seconds: (this.state.seconds - 1)
    });

    if (this.state.seconds === 0){
      console.log('time up');
      clearInterval(this.timer);
      this.props.shouldRestart();

    }

  }

  render() {
    return (
      <div className='timer-container'>
        <h2 className='timer-clock'>{this.state.seconds}</h2>
        <button className='start-btn' onClick={this.startTimer}>Start Timer</button>
        <button className='restart-btn' onClick={this.restartTimer}>Restart Button</button>
      </div>
    );
  }
}
  




