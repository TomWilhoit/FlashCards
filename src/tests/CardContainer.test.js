import React from 'react';
import Timer from '../Timer';
import { shallow } from 'enzyme';
import CardContainer from '../CardContainer'
import App from '../App'
import FlashCardDataSet from '../FlashCardDataSet'

const incrementQuestionIndexMock = jest.fn()
const saveToStorageMock = jest.fn()
const finishGameMock = jest.fn()

describe('CardContainer', () => {
  let wrapper;
  
  beforeEach(() => {
  wrapper = shallow(
    <CardContainer
        questions= {FlashCardDataSet}
        incrementQuestionIndex = {incrementQuestionIndexMock}
        saveToStorage = {saveToStorageMock}
        questionIndex = {0}
        shouldRestart={false}
        finishGame={finishGameMock}/>
    );
  });
  
  it('should match snapshot when all data is passed correctly', () => {
    expect(wrapper).toMatchSnapshot();
  }); 
});
