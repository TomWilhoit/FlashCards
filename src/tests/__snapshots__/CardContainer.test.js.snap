// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should match snapshot when all data is passed correctly 1`] = `
<div>
  <Card
    correctAnswer="The value of the this keyword, in reference to the object that owns the currently executing code"
    falseAnswer1="Context is the exact same as scope"
    falseAnswer2="Function based reference to the function that 'owns' the currently executing code"
    finishGame={[MockFunction]}
    id={1}
    incrementQuestionIndex={[MockFunction]}
    question="What is 'context' in Javascript?"
    saveToStorage={[MockFunction]}
    shouldRestart={false}
  />
</div>
`;
